### DISCUSSION TITLE
Is it possible to configure mini.pairs to insert a pair of parentheses after a function is inserted from the completion menu?

### DISCUSSION THREAD

**Discussion by jarmos-san** [nvim-mini/mini.nvim#399] [Q&A] [answered]
I use [hrsh7th/nvim-cmp](https://github.com/hrsh7th/nvim-cmp) for auto-completion and was it with another plugin - [windwp/nvim-autopairs](https://github.com/windwp/nvim-autopairs) to insert a bracket pair after entering a function from the completions menu. The functionality is well documented in [nvim-cmp's wiki](https://github.com/hrsh7th/nvim-cmp/wiki/Advanced-techniques#add-parentheses-after-selecting-function-or-method-item).

So, is it possible to configure mini.pairs and nvim-cmp to behave the same?

**Comment by echasnovski** [comment, marked as answer]
> I use [hrsh7th/nvim-cmp](https://github.com/hrsh7th/nvim-cmp) for auto-completion and was it with another plugin - [windwp/nvim-autopairs](https://github.com/windwp/nvim-autopairs) to insert a bracket pair after entering a function from the completions menu. The functionality is well documented in [nvim-cmp's wiki](https://github.com/hrsh7th/nvim-cmp/wiki/Advanced-techniques#add-parentheses-after-selecting-function-or-method-item).
> 
> So, is it possible to configure mini.pairs and nvim-cmp to behave the same?

Functionality of 'mini.pairs' when it comes to actually inserting pairs is quite straightforward: it creates mapings for opening parts to insert whole pair instead of just opening symbol. Hence there is no special functions to emit pairs.

Based on [implementation of `on_confirm_done()`](https://github.com/windwp/nvim-autopairs/blob/master/lua/nvim-autopairs/completion/cmp.lua) from 'nvim-autopairs', it seems to get necessary information from 'nvim-cmp' event data and act accordingly.

Sorry, but configuring compatibility with other tools is out of scope. Mostly because of expected instability of interfaces ~~and presence of 'mini.completion'~~.

**Reply by jarmos-san** [reply]
Ah makes sense now! No worries about not supporting certain functionalities which requires compatibility with other tools. The minimalism provided by all of the mini.nvim libraries is the primary reason for me to choose those over the alternatives. So, yes I don't mind compromising on a trivial functionality.
